### DISCUSSION TITLE
Notification for unloaded tiles?

### DISCUSSION THREAD

**Discussion by wangfeng1981** [pelicanmapping/osgearth#2512] [Support Q&A] [answered]
Hi guys, are there APIs for detecting which tiles are unloaded in the scene? I want to add Bullet physics engine for the terrain. I use a callback of onTileUpdated for adding the tile rigid body. However I do not when and how to remove those rigid body of unloaded tiles. Thanks in advance.

**Comment by gwaldron** [comment, marked as answer]
No; there is no callback for a removed tile.
However, onTileUpdate() passes you a node. If you store an osg::observer_ptr<> for this node, you should be able to detect when it gets deleted.
